clippy_lints/src/single_component_use_path.rs

```diff
@@ -0,0 +1,51 @@
+use crate::utils::span_lint_and_sugg;
+use rustc_lint::{EarlyLintPass, EarlyContext};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use syntax::ast::{Item, ItemKind};
+use if_chain::if_chain;
+use rustc_errors::Applicability;
+
+declare_clippy_lint! {
+    /// **What it does:** Checking for imports with single component use path.
+    ///
+    /// **Why is this bad?** Import with single component use path such as `use cratename;`
+    /// is not necessary in order to use the crate.
+    ///
+    /// **Known problems:** None.
+    ///
+    /// **Example:**
+    /// 
+    /// ```rust, ignore
+    /// use bit_vec;
+    ///
+    /// fn main() {
+    ///     bit_vec::BitVec::new();
+    ///}
+    ///```
+    pub SINGLE_COMPONENT_USE_PATH,
+    style,
+    "use with single component path is redundant"
+}
+
+declare_lint_pass!(SingleComponentUsePath => [SINGLE_COMPONENT_USE_PATH]);
+
+impl EarlyLintPass for SingleComponentUsePath {
+    fn check_item(&mut self, cx: &EarlyContext<'_>, item: &Item) {
+        if_chain! {
+            if let ItemKind::Use(use_tree) = &item.kind;
+            if let segments = &use_tree.prefix.segments;
+            if segments.len() == 1;
```

Comment:
I forget to check the `use ident as other;` case, adding this as a reminder.

Reply:
That shouldn't lint, right?

Reply:
No, it shouldn't.